### DISCUSSION TITLE
React Native & Stripe integration

### DISCUSSION THREAD

**Discussion by ceceppa** [swellstores/community#416] [Help] [answered]
I'm trying to integrate the Stripe SDK with Swell, but I need to figure out how to pass the information from Stripe to Swell.
What I've tried so far:

1. Trigger the PaymentIntent on Stripe
2. Pass the Stripe payment intent ID to Swell:

```
const billing = {
  intent: {
    stripe: {
      id: '<payment_intent_id>'
    }
  }
}

// Using Swell JS library
await swell.cart.update({ billing })
```

like described here: https://developers.swell.is/frontend-api/payments#create-a-payment-method

3. call `createOrder`

I can see the order being created on Swell admin page, but:

- Status is: Unpaid
- Payments section is empty

What's the correct way to attach the payment to the order, so that Swell can finalise it?

**Comment by sags95** [comment, marked as answer]
Hey @ceceppa I've detailed a similar response here: https://github.com/orgs/swellstores/discussions/299#discussioncomment-6502406

It's not exactly for the React Native SDK but the details are mainly around creating paymentIntents with the correct params and updating the cart.
